yapapi/rest/activity.py

```diff
@@ -84,35 +84,26 @@ async def __aexit__(self, exc_type, exc_val, exc_tb) -> None:
         # on destroy_activity event.
         if exc_type:
             _log.debug(
-                "activity %s CLOSE for [%s] %s",
-                self._id,
-                exc_type.__name__,
-                exc_val,
-                exc_info=(exc_type, exc_val, exc_tb),
+                "Closing activity %s on error:", self._id, exc_info=(exc_type, exc_val, exc_tb)
             )
+        else:
+            _log.debug("Closing activity %s", self._id)
         try:
             batch_id = await self._api.call_exec(
                 self._id, yaa.ExeScriptRequest(text='[{"terminate":{}}]')
             )
-            with contextlib.suppress(yexc.ApiException, asyncio.TimeoutError):
+            with contextlib.suppress(asyncio.TimeoutError):
                 # wait 1sec before kill
-                await self._api.get_exec_batch_results(self._id, batch_id, timeout=1.0)
-        except yexc.ApiException as err:
-            # Suppress errors that say that this activity does not exist (we're closing it anyway).
-            msg_to_suppress = (
-                f"No service registered under given address '/public/exeunit/{self._id}/Exec'"
-            )
-            level = (
-                logging.ERROR
-                if err.status != 500 or msg_to_suppress not in err.body
-                else logging.DEBUG
-            )
-            _log.log(level, "failed to destroy activity: %s", self._id, exc_info=True)
+                await asyncio.wait_for(
```

Comment:
Use `wait_for` to enforce the 1s timeout

Reply:
why 1s in particular?

Reply:
I think (I'm not the original author) that sending this "terminate" command was just "w/o for some buggy providers which do not kill exe-unit" as the comment on top of the function body states. So we were sending "terminate" just in case, and didn't want to wait for it to complete, hence just 1s timeout. But somehow the timeout didn't work when interrupting the activity (vs normal closing), perhaps because the provider was busy executing other commands? So it just hang there until cancelled from the outside.

Reply:
Anyway, 1s timeout was already there, specified for the `self._api.get_exec_batch_results()` function, only it didn't always work, so I wrapped it with `asyncio.wait_for(..., timeout=1.0)`.